### PR TITLE
Updates for Adversarial Patch attack in TensorFlow

### PR DESCRIPTION
**Pull request by beat-buesser**
# Description

This pull request adds support for models predicting logits in addition to existing support for models predicting probability.

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by minhitbk — APPROVED** [review]
Look good. A small question, why don't you check the probability in the __init__ with some faked x and assign use_logits to True/False just once, instead of doing it for every generate?

**Comment by beat-buesser** [comment]
Hi @minhitbk Thank you. Actually, I haven't thought if it. Let me try to move the check to `__init__`. I think it will require extra code to handle wild cards in the input shape.
